client/request.go

```diff
@@ -1,44 +1,40 @@
 package client
 
 import (
-	"encoding/binary"
-
 	rt "github.com/ssmccoy/g2/pkg/runtime"
 )
 
 // Request from client
 type request struct {
-	DataType rt.PT
-	Data     []byte
+	pt   rt.PT
+	data [][]byte
```

Comment:
Is Data still in use? It is basically `[]byte`, and you replace `Data []byte` with `data [][]byte` here.

Reply:
No I removed `Data []byte` from the type. Basically, the specification of the protocol is such that the parameters to a specific packet are terminated with `\0`; with the last one generally being the "general payload" [and as such can have nulls].

So when the request requires, for instance `function name`, `id`, `data` it will be "function\0id\0data"; well, it's easier for the callers to simply give me a slice of slices (`[][]byte`) than it to have the callers all worry about their null separators.